osu.Game/Screens/Play/HUDOverlay.cs

```diff
@@ -258,13 +258,13 @@ protected override void Update()
 
             Vector2? highestBottomScreenSpace = null;
 
-            foreach (var element in mainComponents.Components)
-                processDrawable(element);
```

Comment:
So to take a note as a developer, since we're iterating over `Components` while it's an interface type, using a `foreach` loop will implicitly call `GetEnumerator`, and the method will box the internal list enumerator struct to an object, therefore causing memory allocation per call. Do I have that right?

Reply:
that's the case, yep.

Reply:
The issue in this case is the iteration of the `iBindableList`, not the interface type.

See https://github.com/ppy/osu-framework/blob/e10051e6643731e393b09de40a3a3d209a545031/osu.Framework/Bindables/IBindableList.cs#L41-L44.

Reply:
Yeah well, it all boils down to the return type of `GetEnumerator`. That makes sense.